### PR TITLE
[netcore] Simplify Array.Length/LongLength implementation

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: netcore/System.Private.CoreLib/src/System/Array.cs [hunk 1]
```diff
@@ -20,27 +20,9 @@ private class RawData
...
-			get {
-				int length = GetLength (0);
-
-				for (int i = 1; i < Rank; i++) {
-					length *= GetLength (i);
-				}
-				return length;
-			}
-		}
+		public int Length => checked ((int)Unsafe.As<RawData>(this).Count);
```
Comment (EgorBo) [inline netcore/System.Private.CoreLib/src/System/Array.cs]: I guess you will need to remove the intrinsic then 🙂

Comment (filipnavara) [reply netcore/System.Private.CoreLib/src/System/Array.cs]: Oh right.

Comment (jaykrell) [reply netcore/System.Private.CoreLib/src/System/Array.cs]: Why only netcore?

Comment (filipnavara) [reply netcore/System.Private.CoreLib/src/System/Array.cs]: The motivation for this was https://github.com/dotnet/coreclr/pull/27123. I was testing the grounds to try to see if we can share more of the code with CoreCLR eventually. The only reason to port it to non-netcore is code parity and removing the native code altogether. It doesn't generate better or worse code in the end so I am not sure if it's worth it.

**Comment by lambdageek** [comment]
I want to hear what @vargaz thinks, but I think it's worth moving this to managed in non-netcore Mono and cleaning up the unmanaged code.

**Comment by marek-safar** [comment]
Closing as we have already intrinsic for this code paths

**Pull request closed without merging**